app/src/main/java/org/luxoft/sdl_core/BluetoothLongWriter.java

```diff
@@ -59,12 +63,25 @@ public void onLongMessageSent() {
         triggerNextMessageProcessing();
     }
 
-    public void processWriteOperation(final byte[] message) {
-        Log.d(TAG, "Going to send message of size " + message.length);
-
+    public void processWriteOperation(final byte[] message_) {
         final int max_message_size = mMtu - mReserved;
+        final boolean need_compress = message_.length >= max_message_size;
+        Log.d(TAG, "Going to send message of size " + message_.length + "; compress " + need_compress);
+
+        byte[] message = message_;
+
+        if(need_compress) {
+            try {
+                message = CompressionUtil.compress(message_);
+                Log.d(TAG, "Compressed size " + message.length);
+            } catch (IOException e) {
+                message = message_;
+                e.printStackTrace();
+            }
+        }
```

Comment:
@VadymLuchko can be a separate function like in long reader

Reply:
done in 568f87bcadda8f2ec653f1aa5a8651fa787e7d8c